src/main/kotlin/com/exactpro/th2/check1/CollectorService.kt

```diff
@@ -90,11 +90,11 @@ class CollectorService(
     @Throws(InterruptedException::class)
     fun verifyCheckRule(request: CheckRuleRequest): ChainID {
         val chainID = request.getChainIdOrGenerate()
-        val task = ruleFactory.createCheckRule(request)
 
         cleanupTasksOlderThan(olderThanDelta, olderThanTimeUnit)
 
-        eventIdToLastCheckTask.compute(CheckTaskKey(chainID, request.connectivityId)) { _, value ->
+        eventIdToLastCheckTask.compute(CheckTaskKey(chainID, request.connectivityId)) { key, value ->
+            val task = ruleFactory.createCheckRule(request, eventIdToLastCheckTask.containsKey(key))
```

Comment:
You already know about value for the key because you are computing one
```suggestion
            val task = ruleFactory.createCheckRule(request, value != null)
```